File: src/Neuralm.Presentation.Web/src/interfaces/ITrainingSessionService.ts
import StartTrainingSessionRequest from '../messages/requests/StartTrainingSessionRequest';
import StartTrainingSessionResponse from '../messages/responses/StartTrainingSessionResponse';
import EndTrainingSessionRequest from '../messages/requests/EndTrainingSessionRequest';
import EndTrainingSessionResponse from '../messages/responses/EndTrainingSessionResponse';
import GetOrganismsRequest from '../messages/requests/GetOrganismsRequest';
import GetOrganismsResponse from '../messages/responses/GetOrganismsResponse';

/**
 * Represents the training session service interface.
 */
export default interface ITrainingSessionService {
  /**
   * Starts the training session.
   * @param startTrainingSessionRequest The start training session request object.
   * @returns The start training session response.
   */
  startTrainingSession(startTrainingSessionRequest: StartTrainingSessionRequest): Promise<StartTrainingSessionResponse>;

  /**
   * Ends the training session.
   * @param endTrainingSessionRequest The end training session request object.
   * @returns The end training session response.
   */
  endTrainingSession(endTrainingSessionRequest: EndTrainingSessionRequest): Promise<EndTrainingSessionResponse>;

  /**
   * Gets the organisms from the session.
   * @param getOrganismsRequest The get organisms request object.
   * @returns The get organisms response.
   */
  getOrganisms(getOrganismsRequest: GetOrganismsRequest): Promise<GetOrganismsResponse>;
}

File: src/Neuralm.Services/Neuralm.Services.TrainingRoomService/Neuralm.Services.TrainingRoomService.Application/Interfaces/ITrainingSessionService.cs
using System.Threading.Tasks;
using Neuralm.Services.Common.Application.Interfaces;
using Neuralm.Services.TrainingRoomService.Messages;
using Neuralm.Services.TrainingRoomService.Messages.Dtos;

namespace Neuralm.Services.TrainingRoomService.Application.Interfaces
{
    /// <summary>
    /// Represents the <see cref="ITrainingSessionService"/> interface.
    /// </summary>
    public interface ITrainingSessionService : IService<TrainingSessionDto>
    {
        /// <summary>
        /// Starts a training session asynchronously.
        /// </summary>
        /// <param name="startTrainingSessionRequest">The start training session request.</param>
        /// <returns>Returns an awaitable <see cref="Task"/> with parameter type <see cref="StartTrainingSessionResponse"/>.</returns>
        Task<StartTrainingSessionResponse> StartTrainingSessionAsync(StartTrainingSessionRequest startTrainingSessionRequest);

        /// <summary>
        /// Ends a training session asynchronously.
        /// </summary>
        /// <param name="endTrainingSessionRequest"></param>
        /// <returns>Returns an awaitable <see cref="Task"/> with parameter type <see cref="EndTrainingSessionResponse"/>.</returns>
        Task<EndTrainingSessionResponse> EndTrainingSessionAsync(EndTrainingSessionRequest endTrainingSessionRequest);

        /// <summary>
        /// Gets the organisms for a training session asynchronously.
        /// </summary>
        /// <param name="getOrganismsRequest">The get organism request.</param>
        /// <returns>Returns an awaitable <see cref="Task"/> with parameter type <see cref="GetOrganismsResponse"/>.</returns>
        Task<GetOrganismsResponse> GetOrganismsAsync(GetOrganismsRequest getOrganismsRequest);

        ///// <summary>
        ///// Posts the scores of the organisms for a training session asynchronously.
        ///// </summary>
        ///// <param name="postOrganismsScoreRequest">The post organisms score request.</param>
        ///// <returns>Returns an awaitable <see cref="Task"/> with parameter type <see cref="PostOrganismsScoreResponse"/>.</returns>
        //Task<PostOrganismsScoreResponse> PostOrganismsScoreAsync(PostOrganismsScoreRequest postOrganismsScoreRequest);
    }
}
